lib/fs.js

```diff
@@ -1562,6 +1562,8 @@ fs.unwatchFile = function(filename, listener) {
   }
 };
 
+// Cache for JS real path 
+var realpathCache = {};
```

Comment:
That's a thing that needs to be solved. One global cache is probably not a good idea. But some caching is needed for performance reasons.

I would guess user could add cache object to options. It could be also used on Linux, to cache  realpath results (not the partial ones as in JS implementation)